### PR TITLE
V2 api fix (denorm fix)

### PR DESCRIPTION
**Pull request by yashashkumar**

### CODE REVIEW

File: api-service/src/controllers/DatasetRead/DatasetRead.ts [hunk 1]
```diff
@@ -34,7 +35,14 @@ const datasetRead = async (req: Request, res: Response) => {
     const { fields, mode } = req.query;
     const attributes = !fields ? defaultFields : _.split(<string>fields, ",");
     const dataset = (mode == "edit") ? await readDraftDataset(dataset_id, attributes) : await readDataset(dataset_id, attributes)
-    if(!dataset) {
+    if (dataset.connectors_config) {
+        dataset.connectors_config = dataset.connectors_config.map((connector: any) => ({
```
Comment (SanthoshVasabhaktula) [inline api-service/src/controllers/DatasetRead/DatasetRead.ts:39]: Wouldn't this throw an exception if the dataset is undefined. Shouldn't it be in the else block before the response is sent?

File: api-service/src/services/DatasetService.ts [hunk 2]
```diff
@@ -260,8 +280,9 @@ class DatasetService {
                 await this.createDruidDataSource(draftDataset, transaction);
             }
             if(indexingConfig.lakehouse_enabled) {
-                const liveDataset = await this.getDataset(draftDataset.dataset_id, ["id"], true);
-                if(liveDataset) {
+                const liveDataset = await this.getDataset(draftDataset.dataset_id, ["id", "api_version"], true);
+
+                if(liveDataset && liveDataset.api_version === "v2") {
```
Comment (SanthoshVasabhaktula) [inline api-service/src/services/DatasetService.ts:285]: Why is the api_version === v2 check required here. If it is not v2 we would have rejected it during validation itself right?

**Review by SanthoshVasabhaktula — CHANGES_REQUESTED** [review]

File: api-service/src/controllers/DatasetRead/DatasetRead.ts (hunk 1 above)
Comment (yashashkumar) [reply api-service/src/controllers/DatasetRead/DatasetRead.ts:42]: Throwing dataset not found error if dataset is null before execution of logic

File: api-service/src/services/DatasetService.ts (hunk 2 above)
Comment (yashashkumar) [reply api-service/src/services/DatasetService.ts:287]: v1 datasets dosn't have the hudi spec to update, so if the dataset is v2 then only we are allowing it to update

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Pull request merged**